### PR TITLE
feature/ICL-66 authorize patron

### PR DESCRIPTION
**Pull request by JGiter**
To verify that patron is authorized to stake iam-lib need to know patron required roles. This PR extends StakingPool entity with `patronRoles` column alogn with `minStakingPeriod`, `withdrawDelay` and `patronRewardPortion` pool constructor parameters
- [x] Updating of existing pool will not throw
- [x] Added endpoint to get pool
- [x] Identify pool by address

### CODE REVIEW

File: src/common/isBigNumber.validator.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import {
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+  ValidationArguments,
+} from 'class-validator';
+
+@ValidatorConstraint({ name: 'isBigNumber', async: false })
+export class IsBigNumber implements ValidatorConstraintInterface {
+  validate(val: unknown, args: ValidationArguments) {
+    return val instanceof BigNumber;
```
Comment (Harasz) [inline src/common/isBigNumber.validator.ts:11]: I think we should add more asserts here. Checking for format of the `val` or something

File: src/modules/claim/entities/claim.entity.ts [hunk 2]
```diff
@@ -3,18 +3,23 @@ import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
 import { IClaim } from '../claim.types';
 import { JWT } from '@ew-did-registry/jwt';
 import { Keys } from '@ew-did-registry/keys';
+import { BadRequestException } from '@nestjs/common';
```
Comment (Harasz) [inline src/modules/claim/entities/claim.entity.ts:6]: Import order

File: src/modules/staking/dtos/staking.pool.dto.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { IsArray, IsString, Validate } from 'class-validator';
+import { BigNumber } from 'ethers';
+import { IsBigNumber } from '../../../common/isBigNumber.validator';
+import { StakingTermsDTO } from './staking.terms.dto';
+
+export class StakingPoolDTO {
+  @IsString()
+  address: string;
```
Comment (Harasz) [inline src/modules/staking/dtos/staking.pool.dto.ts:8]: Don't we miss here `@ApiPropert()` decorator from `@nestjs/swagger`?

File: src/modules/staking/staking.service.ts [hunk 4]
```diff
@@ -12,6 +12,11 @@ import { Logger } from '../logger/logger.service';
 import { StakingPoolFactory__factory } from '../../ethers/factories/StakingPoolFactory__factory';
 import { StakingPoolFactory } from '../../ethers/StakingPoolFactory';
 import { Provider } from '../../common/provider';
+import { RoleService } from '../role/role.service';
+import { ConfigService } from '@nestjs/config';
+import { SchedulerRegistry } from '@nestjs/schedule';
```
Comment (Harasz) [inline src/modules/staking/staking.service.ts:17]: Import order

File: src/modules/staking/staking.service.ts [hunk 5]
```diff
@@ -21,15 +26,33 @@ export class StakingService implements OnModuleDestroy {
     private readonly stakingTermsRepository: Repository<StakingTerms>,
     @InjectRepository(StakingPool)
     private readonly stakingPoolRepository: Repository<StakingPool>,
+    private readonly roleService: RoleService,
+    private readonly orgService: OrganizationService,
     private readonly logger: Logger,
     private readonly provider: Provider,
+    private config: ConfigService,
+    private schedulerRegistry: SchedulerRegistry,
```
Comment (Harasz) [inline src/modules/staking/staking.service.ts:34]:
```suggestion
    private readonly schedulerRegistry: SchedulerRegistry,
```

File: src/modules/staking/staking.service.ts [hunk 6]
```diff
@@ -21,15 +26,33 @@ export class StakingService implements OnModuleDestroy {
     private readonly stakingTermsRepository: Repository<StakingTerms>,
     @InjectRepository(StakingPool)
     private readonly stakingPoolRepository: Repository<StakingPool>,
+    private readonly roleService: RoleService,
+    private readonly orgService: OrganizationService,
     private readonly logger: Logger,
     private readonly provider: Provider,
+    private config: ConfigService,
```
Comment (Harasz) [inline src/modules/staking/staking.service.ts:33]:
```suggestion
    private readonly config: ConfigService,
```

**Review by Harasz — CHANGES_REQUESTED** [review]
Need changes

**Comment by JGiter** [comment]
Obsolete because of https://github.com/energywebfoundation/iam-client-lib/blob/develop/src/modules/staking/staking-pool.service.ts

**Pull request closed without merging**